kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/channel/Channel.java

```diff
@@ -142,20 +210,60 @@ public boolean isEnabled() {
      *
      * @return a double that represents the scale factor to be applied to the read value
      *
+     * @throws IllegalArgumentException
+     * 
      * @since 2.3
+     * 
+     * @deprecated Use {@link #getValueScaleAsTypedValue()}
      */
+    @Deprecated
     public double getValueScale() {
+        if (this.scaleOffsetType != ScaleOffsetType.DEFINED_BY_VALUE_TYPE) {
```

Comment:
As per previous comment, is it better to return `this.valueScale.doubleValue()` instead of throwig?

Reply:
it's an illegal state of the object, i would throw the exception  because using the method results in an unintentional casting

Reply:
i'm reconsidering this, an user needs to change the api in a client application to not break it loading a configuraiton without the scaleoffset value property, i'll remove the exception throwing